### PR TITLE
Filter string: IN operator added

### PR DESCRIPTION
**Pull request by vicilliar**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
IN operator for vespa querying is not supported.

* **What is the new behavior (if this is a feature change)?**
IN operator is supported in structured indexes only. It also only works on `string` and `int` vespa fields.

USER INPUT: List must be surrounded by `()`. No quotes will be used for list items. Marqo will internally determine whether to treat these like an int or a string. Internal list whitespaces are ignored.

Examples:
```
a in (1,2,3)
a in (b,c,d)
a in (1,   2,   3)
```
* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
In progress.

* **Related Python client changes** (link commit/PR here)

* **Related documentation changes** (link commit/PR here)

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/core/search/search_filter.py [hunk 1]
```diff
@@ -6,14 +6,16 @@
 from marqo.exceptions import InternalError
 
 
+IN_TERM_DIVIDER = ' IN ('
```
Comment (wanliAlex) [inline src/marqo/core/search/search_filter.py]: Maybe you can move this to the class `MarqoFilterStringParser` if it is only used there

File: src/marqo/core/search/search_filter.py [hunk 2]
```diff
@@ -207,6 +231,50 @@ class MarqoFilterStringParser:
     class _TermType(Enum):
         Equality = 1
         Range = 2
+        In = 3
+
+    def _term_divider_is_IN(self, i: int, filter_string: str):
```
Comment (wanliAlex) [inline src/marqo/core/search/search_filter.py]: nit: return type

File: src/marqo/core/structured_vespa_index/structured_vespa_index.py [hunk 3]
```diff
@@ -453,6 +453,49 @@ def _get_tensor_search_term(self, marqo_query: MarqoTensorQuery) -> str:
         else:
             return ''
 
+    def convert_to_in_list_str(self, value_list: list, marqo_field_name: str, marqo_field_type: FieldType) -> str:
```
Comment (wanliAlex) [inline src/marqo/core/structured_vespa_index/structured_vespa_index.py]: Add a `_` to indicate it's for internal class use

Comment (wanliAlex) [inline src/marqo/core/structured_vespa_index/structured_vespa_index.py]: You may also want to move this function under `_get_filter_term()`, like `escape`. Ask Farshid to confirm on this.

File: tests/core/search/test_search_filter.py [hunk 4]
```diff
@@ -144,7 +305,15 @@ def test_parse_malformedString_fails(self):
             (')', ')'),
             ('()', '()'),
             ('a:1 AND (b:2 OR (c:3 AND (d:4 OR e:5)) OR d:6', 'imbalanced parentheses, not closed'),
-            ('a:1 AND b:2 OR (c:3 AND (d:4 OR e:5))) OR d:6', 'imbalanced parentheses, not opened')
+            ('a:1 AND b:2 OR (c:3 AND (d:4 OR e:5))) OR d:6', 'imbalanced parentheses, not opened'),
+
+            # In term tests
```
Comment (wanliAlex) [inline tests/core/search/test_search_filter.py:310]: I can think of more edge cases like `[ ]` inside `( )`. Use your judgement to see if these cases should be tested or not. Maybe chatgpt can help you generate these edge cases quickly.

**Review by wanliAlex — CHANGES_REQUESTED** [review]

File: src/marqo/core/search/search_filter.py (hunk 1 above)
Comment (vicilliar) [reply src/marqo/core/search/search_filter.py]: Done

File: src/marqo/core/search/search_filter.py (hunk 2 above)
Comment (vicilliar) [reply src/marqo/core/search/search_filter.py]: Done

File: src/marqo/core/structured_vespa_index/structured_vespa_index.py (hunk 3 above)
Comment (vicilliar) [reply src/marqo/core/structured_vespa_index/structured_vespa_index.py]: Done

Comment (vicilliar) [reply src/marqo/core/structured_vespa_index/structured_vespa_index.py]: Done

File: tests/core/search/test_search_filter.py (hunk 4 above)
Comment (vicilliar) [reply tests/core/search/test_search_filter.py:310]: Added an edge case! Can't think of any other meaningful tests for this.

**Review by wanliAlex — APPROVED** [review]
